data/index.js

```diff
@@ -24,13 +24,70 @@ const selectors = {};
 const actions = {};
 let listeners = [];
 
+/**
+ * Higher-order function creator to create a function which invokes ifTrueFn
+ * only if the given predicate returns a truthy value.
+ *
+ * @param {Function} predicate Predicate test.
+ *
+ * @see http://ramdajs.com/docs/#when
+ *
+ * @return {Function} Higher-order function to call with ifTrueFn.
+ */
+const when = ( predicate ) => ( ifTrueFn ) => ( ...args ) => {
```

Comment:
It seems like `when` and `whenWith` should be extracted to their own file so we could have only public APIs in this file. Can we also add simple unit tests for both functions?

Reply:
Create our own FP utility library? 😄

Reply:
I'm fine about pulling `ramda` and `recompose` in :D

Reply:
![Disappear](https://i.imgur.com/xk7EfxI.gif)

Reply:
Since 84c64c8 was removed in rebase, this is no longer relevant.